spec/var-ref.test.sh

```diff
@@ -76,10 +77,13 @@ A_keys=0
 A_nobrackets=0
 ## END
 
-#### ${!a[@]-'default'} is illegal
+#### ${!a[@]-'default'} is legal but fails with more than one element
 
-# bash disallows this when a is an array.  We make it an error because [@]
-# implies it's an array.
+# bash allows this construct, but the indirection fails when the array has more
+# than one element because the variable name contains a space.  OSH originally
+# made it an error unconditionally because [@] implies it's an array, so the
+# behavior has been different from Bash when the array has a single element.
+# We now changed it to follow Bash even when the array has a single element.
```

Comment:
BTW is is possible now to fail in the middle of a word, and then `return 1` for the command that the word belongs to

That is done with FailGlob

I think that is what bash is doing

I am still a bit confused about the test case, but maybe we can wait for the bash 5 change to see if we want to do that or not

Reply:
#2204